test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/CleanProjectIntegrationTest.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class CleanProjectIntegrationTest : MSBuildIntegrationTestBase
+    {
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task CleanProject_RunBuild()
+        {
+            var result = await DotnetMSBuild("Restore;Build"); // Equivalent to dotnet build
```

Comment:
You'll likely hit issues on CI with this because CI manipulates CI restore settings. In other repos, we workaround this by preserving those settings as a compiled artifact so the tests can use it. For example:

- DotNetRestoreSources
- DotNetPackageVersionPropsPath
- DotNetBuildOffline

Example: aspnet/Scaffolding handles this by generating a C# file.
Template:
https://github.com/aspnet/Scaffolding/blob/3175fc45a6d3491fce6d6824d6a2ff8792a3ec3b/test/Shared/MsBuildProjectStrings.cs.in#L16
Target:
https://github.com/aspnet/Scaffolding/blob/dev/test/Directory.Build.targets#L23-L67

Reply:
I don't need this to **look like** a real site, I just need to compile. Would it be better for me to just reference binaries directly? I have a few small number of dependencies.

Reply:
To clarify this, the project that I'm generating here has a fixed set of dependencies. It has a p2p reference to some razor binaries, but it doesn't depend on **any** ASP.NET Core packages outside of this repo.

Reply:
Talked in person. Since the test app only uses stable, RTM packages, we don't need this complexity for now. We might need it in the future if we wanted tests that run against latest, pre-release corefx packages